sql/src/main/java/io/druid/sql/avatica/DruidStatement.java

```diff
@@ -94,7 +103,7 @@ public DruidStatement(
     this.statementId = statementId;
     this.queryContext = queryContext == null ? ImmutableMap.of() : queryContext;
     this.onClose = Preconditions.checkNotNull(onClose, "onClose");
-    this.yielderOpenCloseExecutor = Executors.newSingleThreadExecutor();
+    this.yielderOpenCloseExecutor = Execs.singleThreaded("JDBCYielderOpenCloseExecutor");
```

Comment:
This executor needs to either be shared between statements, or shutdown when the statement is closed, otherwise there will be a bunch of dangling threads.

If it remains owned by the statement, it should have the statement id somewhere in the name, so you can understand what's going on in a thread dump.